compiler/src/iree/compiler/Codegen/Sandbox/LinalgTensorCodegenDriver.cpp

```diff
@@ -368,6 +419,135 @@ void LinalgFusePass::runOnOperation() {
     return signalPassFailure();
   }
 }
+namespace {
+/// Pattern to wrap splitReduction transformation.
+struct CodegenSplitReduction
+    : public OpInterfaceRewritePattern<linalg::LinalgOp> {
+  CodegenSplitReduction(MLIRContext *context, bool reductionReordering,
+                        int64_t size, uint64_t unrollFactor,
+                        linalg::LinalgTransformationFilter filter,
+                        PatternBenefit benefit = 1)
+      : OpInterfaceRewritePattern<linalg::LinalgOp>(context, benefit),
+        reductionReordering(reductionReordering),
+        size(size),
+        unrollFactor(unrollFactor),
+        filter(std::move(filter)) {}
+
+  LogicalResult matchAndRewrite(linalg::LinalgOp op,
+                                PatternRewriter &rewriter) const override {
+    // Make sure that only generic ops are supported, only one input and the
+    // innermost tensor dimension is reduction dimension.  Everything else is
+    // copied from the original SplitReduction Transformation.
+    // TODO: support named ops, numInputs > 1, and modify lastDim check below
+    // accordingly. If reductionReordering is not enabled by default, it must be
+    // an integer or index type to proceed to allow associative reordering.
+    if (failed(filter.checkAndNotify(rewriter, op)) ||
+        !op.hasTensorSemantics() || op.getNumReductionLoops() != 1 ||
+        op.getNumOutputs() != 1 || !op.hasOnlyProjectedPermutations() ||
+        !isa<linalg::GenericOp>(op) || op.getNumInputs() != 1 ||
+        !(reductionReordering || op.getInputOperand(0)
+                                     ->get()
+                                     .getType()
+                                     .dyn_cast<ShapedType>()
+                                     .getElementType()
+                                     .isIntOrIndex()))
```

Comment:
Take a look at the current splitReduction code here. https://github.com/llvm/llvm-project/blob/81ca5aa452400843235e058bc9c83fe71eccd593/mlir/lib/Dialect/Linalg/Transforms/SplitReduction.cpp#L65

I am exposing exactly those conditions + 3 more.

Reply:
Exposing the SplitReduction pass's preconditions is useless now because I am going to remove most of those preconditions in my next change to splitReduction - I don't want to muck around with LLVM changes again because it looks like it will take an enormous amount of time to get that synced to IREE (given last week's experience).

Reply:
Some of these preconditions are necessary for the IREE pass also: ( !op.hasTensorSemantics() || op.getNumReductionLoops() != 1 ||
        op.getNumOutputs() != 1 || !op.hasOnlyProjectedPermutations())

Even if these preconditions are supported by the base SplitReduction, we are not going to support it in IREE. I can adjust the comment to reflect that we need them.